.github/workflows/pr-checks-build-images.yaml

```diff
@@ -0,0 +1,70 @@
+name: PR checks - Build operator images
+
+on:
+  pull_request:
+    branches:
+      - master
+
+env:
+  SDK_VERSION: "1.3.0"
+  OPM_VERSION: "1.15.2"
+  CONTAINER_RUNTIME: "podman"
+  ARTIFACTS: "artifacts"
+  REGISTRY_PREFIX: quay.io/redhat-developer
+  REPO: servicebinding-operator
+
+jobs:
+  build-operator-images:
+    name: "Build operator images (PR)"
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout Git Repository
+        uses: actions/checkout@v2
+
+      - name: Setup CLI
+        uses: ./.github/actions/setup-cli
+        with:
+          operator-sdk: true
+          opm: true
+
+      - name: Setup local registry
+        run: |
+          .github/actions/setup-podman/setup-local-registry.sh
+          .github/actions/setup-podman/setup-podman-wrapper.sh
+
+      - name: Build images
+        env:
+          PR_NUMBER: ${{github.event.pull_request.number}}
+          PR_SHA: ${{github.event.pull_request.head.sha}}
+          OPERATOR_REPO_REF: ${{env.REGISTRY_PREFIX}}/${{env.REPO}}
+        run: |
+          export TAG=pr-${PR_NUMBER}-${PR_SHA:0:8}
```

Comment:
IMHO, we should not have PR_SHA as the part of the tag - if we are adding commits to PR before merge and while testing, we are going to produce a number of images, but only the last one after merge will be removed. `TAG=pr-${PR_NUMBER}` will override images created by past commits on the same branch/PR.

Reply:
Actually, the `Clean PR images` step deletes all the images that starts with `pr-<PR_NUMBER>-` - not only the last one, so all the images related to the PR are deleted.

Additionally, I correlate and ensure the correct image built from the matching commit using the PR and SHA at OpenShift CI side...

Reply:
ah, thanks, I missed that.